### DISCUSSION TITLE
Docker Image (docker-compose.yml) Error | UNABLE_TO_VERIFY_LEAF_SIGNATURE

### DISCUSSION THREAD

**Discussion by ImAbhishekTomar** [apache/superset#18375] [Q&A / Help]
Hi, 

I am trying to build a superset docker image "docker-compose.yml" but it's giving me an error & "docker-compose-non-dev.yml" image is building successfully.

> docker-compose -f docker-compose.yml up -d --build

### ERROR
![Screenshot 2022-01-20 at 4 54 17 PM](https://user-images.githubusercontent.com/6933841/150329790-cef4399c-5862-45a1-9527-bd28600f8ad3.png)

```
☁  superset [master] ⚡  docker-compose -f docker-compose.yml up -d --build
WARNING: The CYPRESS_CONFIG variable is not set. Defaulting to a blank string.
Building superset-websocket
[+] Building 82.5s (8/9)                                                                                                                                                    
 => [internal] load build definition from Dockerfile                                                                                                                   0.0s
 => => transferring dockerfile: 44B                                                                                                                                    0.0s
 => [internal] load .dockerignore                                                                                                                                      0.0s
 => => transferring context: 35B                                                                                                                                       0.0s
 => [internal] load metadata for docker.io/library/node:16                                                                                                             2.3s
 => [internal] load build context                                                                                                                                      0.0s
 => => transferring context: 1.80kB                                                                                                                                    0.0s
 => [1/5] FROM docker.io/library/node:16@sha256:4b0b5c3db44f567d5d25c80a6fe33a981d911cdae20b39d2395be268aea2cb97                                                       0.0s
 => CACHED [2/5] WORKDIR /home/superset-websocket                                                                                                                      0.0s
 => CACHED [3/5] COPY . .                                                                                                                                              0.0s
 => ERROR [4/5] RUN npm ci                                                                                                                                            80.1s
------                                                                                                                                                                      
 > [4/5] RUN npm ci:                                                                                                                                                        
#8 1.019 npm WARN EBADENGINE Unsupported engine {                                                                                                                           
#8 1.020 npm WARN EBADENGINE   package: 'superset-websocket@0.0.1',                                                                                                         
#8 1.020 npm WARN EBADENGINE   required: { node: '^16.9.1', npm: '^7.5.4' },                                                                                                
#8 1.021 npm WARN EBADENGINE   current: { node: 'v16.13.2', npm: '8.1.2' }
#8 1.021 npm WARN EBADENGINE }
#8 79.99 npm ERR! code UNABLE_TO_VERIFY_LEAF_SIGNATURE
#8 79.99 npm ERR! errno UNABLE_TO_VERIFY_LEAF_SIGNATURE
#8 79.99 npm ERR! request to https://registry.npmjs.org/yn/-/yn-3.1.1.tgz failed, reason: unable to verify the first certificate
#8 80.01 
#8 80.01 npm ERR! A complete log of this run can be found in:
#8 80.01 npm ERR!     /root/.npm/_logs/2022-01-20T11_23_34_538Z-debug.log
------
executor failed running [/bin/sh -c npm ci]: exit code: 1
ERROR: Service 'superset-websocket' failed to build : Build failed
```

**Comment by RivkaRoth** [comment]
I have the same error :(

**Comment by ad-m** [comment]
@ImAbhishekTomar @RivkaRoth this looks like local networks hijack. Are you using corporate network (where some weird https-hijack might happen) or home network?

Could you try run on that via affected network:

```bash
docker run -w /media/ docker.io/library/node:16 npm install yn
```

**Comment by ImAbhishekTomar** [comment]
@ad-m  ![Screenshot 2022-01-27 at 2 40 45 PM](https://user-images.githubusercontent.com/6933841/151327774-02485309-5f7e-465a-a566-9801eab8923b.png)

**Comment by katana8911** [comment]
Was getting this exact error as well, turned out to be the inability of the docker container running superset to be able to reach the internet to download and install packages.
If you need a proxy to reach the internet, have a look at the following:  https://docs.docker.com/network/proxy/
after creating the .docker/config.json and putting in the proxy details, and re-running 
"docker-compose -f docker-compose.yml" threw the same error, and then after a min, began installing the packages it needed, and then superset dev was able to start.

Experienced a secondary issue, which was that after superset started, and logging in, there was no dashboard, just a X and a close icon.

This was resolved by doing:
docker-compose down
docker-compose build
docker-compose up

Then the dev container started up as expected with all the example dashboards.

**Comment by ad-m** [comment]
@ImAbhishekTomar The screenshot you have shown clearly indicates that the problem you reported has nothing to do with the Apache Superset project. The Apache Superset has been excluded from the equation and the problem persists. It is most likely related to the modification of network traffic by the network provider.

Contact your local network administrator or local IT administrator for detailed guidance on how you can get uninterrupted Internet access from Docker through your workstation.

**Comment by mebaysan** [comment]
I get the same error. I use Docker on my virtual machine I created by VirtualBox. I set its network as "Bridge". How to deal with this problem?

**Reply by mebaysan** [reply]
I solved the problem by making executable the `docker-compose.yml` file.

`sudo chmod +x docker-compose.yml`
